torch/autograd/variable.py

```diff
@@ -206,20 +206,9 @@ def register_hook(self, hook):
         return handle
 
     def reinforce(self, reward):
-        """Registers a reward obtained as a result of a stochastic process.
-
-        Differentiating stochastic nodes requires providing them with reward
-        value. If your graph contains any stochastic operations, you should
-        call this function on their outputs. Otherwise an error will be raised.
-
-        Parameters:
-            reward(Tensor): Tensor with per-element rewards. It has to match
-                the device location and shape of Variable's data.
-        """
-        if not isinstance(self.grad_fn, StochasticFunction):
-            raise RuntimeError("reinforce() can be only called on outputs "
-                               "of stochastic functions")
-        self.grad_fn._reinforce(reward)
+        raise RuntimeError(
+            "reinforce() was removed. Use torch.distributions instead.\n"
+            "See http://pytorch.org/docs/master/distributions.html")
```

Comment:
Can we link to the specific paragraph that shows how to replace these calls?

Reply:
maybe we can do:

```
See ...
An example of porting from `reinforce()` to using `torch.distributions` is here: https://github.com/pytorch/examples/pull/249/files
```

wdyt adam?

Reply:
I'm fine with anything that will point people **directly** to the snippet that lets them fix their code, but this diff is quite large so it might be hard to parse. Let's say "see <distributions docs>, where in the header you can find an example showing how to implement policy gradient".